From 78677dc6e262f46937d00c3aa52381e4bb198fa5 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Mon, 13 Apr 2026 14:00:00 +0200
Subject: [PATCH] [codec,progressive] fix underflow guard in
 progressive_rfx_quant_sub

Backport of commit 78677dc6e262f46937d00c3aa52381e4bb198fa5.

Made-with: Cursor
---
 libfreerdp/codec/progressive.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libfreerdp/codec/progressive.c b/libfreerdp/codec/progressive.c
index bbcc921..1234567 100644
--- a/libfreerdp/codec/progressive.c
+++ b/libfreerdp/codec/progressive.c
@@ -158,7 +158,7 @@ static INLINE BOOL progressive_rfx_quant_sub(const RFX_COMPONENT_CODEC_QUANT* q1
                                              const RFX_COMPONENT_CODEC_QUANT* q2,
                                              RFX_COMPONENT_CODEC_QUANT* dst)
 {
-	if (q1->HH1 < q2->HL1)
+	if (q1->HL1 < q2->HL1)
 		return FALSE;
 	dst->HL1 = q1->HL1 - q2->HL1; /* HL1 */
 
-- 
2.49.0

